compiler/rustc_trait_selection/src/traits/specialize/specialization_graph.rs

```diff
@@ -48,7 +48,7 @@ trait ChildrenExt<'tcx> {
 impl<'tcx> ChildrenExt<'tcx> for Children {
     /// Insert an impl into this set of children without comparing to any existing impls.
     fn insert_blindly(&mut self, tcx: TyCtxt<'tcx>, impl_def_id: DefId) {
-        let trait_ref = tcx.impl_trait_ref(impl_def_id).unwrap();
+        let trait_ref = tcx.bound_impl_trait_ref(impl_def_id).unwrap().subst_identity();
```

Comment:
`skip_binder`, the specialization graph from the view of users, so they remain bound.